test/router_SUITE.erl

```diff
@@ -49,6 +89,30 @@ basic_routing(_C) ->
     %% we know that 1 and 3 should be dropped, and 2, 4 and 5 handled by a, b and c respectively
     verify([{a, 2}, {b, 4}, {c, 5}]),
     ok.
+     
+update_tables_hidden_components(_C) ->
+    %% Tables as of b076e4a62a8b21188245f13c42f9cfd93e06e6b7
+    create_component_tables([domain, handler, node]),
+
+    ejabberd_router:update_tables(),
+
+    %% Local table is removed and distributed one has new list of attributes
+    false = lists:member(external_component, mnesia:system_info(tables)),
+    [domain, handler, node, is_hidden] = mnesia:table_info(external_component_global, attributes).
+
+update_tables_hidden_components_idempotent(_C) ->
+    AttrsWithHidden = [domain, handler, node, is_hidden],
+    create_component_tables(AttrsWithHidden),
+
+    ejabberd_router:update_tables(),
+
+    %% Local table is not removed and attribute list of distributed one is not changed
```

Comment:
Local table is not removed and the attribute list of the distributed one is not changed